_posts/2020/2020-02-16-all-your-mailer-views-in-one-place.md

```diff
@@ -0,0 +1,61 @@
+---
+title: "All Your Mailer Views in One Place"
+description: "A little separation"
+layout: article
+category: ruby
+image:
+  base: '2020/all-your-mailer-views-in-one-place'
+  alt: "Envelopes"
+  credit:  Joanna Kosinska
+  source: "https://unsplash.com/photos/uGcDWKN91Fs"
+---
+
+One of the greatest things that Rails provides to developers is a sensible set of defaults for _where stuff goes_ in the file hierarchy. The standard approach to `app/models`, `app/mailers`, `app/views`, `config`, etc.,  allows us to arrive at a Rails project and quickly understand where existing code is and where to place our new code.
+
+I’m therefore loathe to deviate much from the defaults, but I make one exception: email view templates. They live in the same part of the folder hierarchy as the web views, e.g. `app/views/whatever_mailer`, even though Mailers are in a different location to Controllers. Even after a decade of Rails I still struggle to find them.
+
+
+## Instead of...
+
+...storing your mailer templates in their default location:
+
+e.g. for `UserMailer`, the mailer is in `app/mailers/user_mailer.rb` and the view templates are in `app/views/user_mailer`.
+
+
+## Use
+
+...a folder/subdirectory within `app/views` or a new “top-level” directory within `app`.
```

Comment:
This is so much clearer to me — I mean, I know I've read this 5 times already but you know what I mean — now that you've set up the problem in advance.

Reply:
SMELLS LIKE VICTORY